staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go

```diff
@@ -37,12 +37,34 @@ var ForeverTestTimeout = time.Second * 30
 // NeverStop may be passed to Until to make it never stop.
 var NeverStop <-chan struct{} = make(chan struct{})
 
-// StartUntil starts f in a new goroutine and calls done once f has finished.
-func StartUntil(stopCh <-chan struct{}, wg *sync.WaitGroup, f func(stopCh <-chan struct{})) {
-	wg.Add(1)
-	go func() {
-		defer wg.Done()
+// Group is an interface to decouple code from sync.WaitGroup.
+type Group interface {
+	Add(delta int)
+	Done()
+}
+
+// StartWithChannelWithinGroup adds 1 to the group, starts f in a new goroutine and calls g.Done once f has finished.
+// stopCh is passed to f as an argument.
+func StartWithChannelWithinGroup(stopCh <-chan struct{}, g Group, f func(stopCh <-chan struct{})) {
```

Comment:
Well, I can rename the parameter but I **want** the name to imply the specific meaning of it and I want to encourage only specific use of `StartWithChannelWithinGroup` function.

Providing a generic function (a function that does not imply the meaning of this parameter) that accepts `<-chan struct{}` does not make much sense. We don't provide and other overloaded versions, why would we provide a `<-chan struct{}` version without implying this specific meaning?

A better approach would be to change all `stopCh <-chan struct{}` in the whole codebase to `context.Context` to be even more specific. See #815. And that is why I wan to add a version that accepts `context.Context`.